posthog/models/feature_flag/flag_matching.py

```diff
@@ -282,66 +310,93 @@ def query_conditions(self) -> Dict[str, bool]:
                             [],
                         )
 
-                person_fields = []
+                person_fields: List[str] = []
 
-                for feature_flag in self.feature_flags:
-                    for index, condition in enumerate(feature_flag.conditions):
-                        key = f"flag_{feature_flag.pk}_condition_{index}"
-                        expr: Any = None
-                        annotate_query = True
-                        if len(condition.get("properties", {})) > 0:
-                            # Feature Flags don't support OR filtering yet
-                            target_properties = self.property_value_overrides
-                            if feature_flag.aggregation_group_type_index is not None:
-                                target_properties = self.group_property_value_overrides.get(
-                                    self.cache.group_type_index_to_name[feature_flag.aggregation_group_type_index], {}
-                                )
-                            expr = properties_to_Q(
-                                Filter(data=condition).property_groups.flat,
-                                override_property_values=target_properties,
+                def condition_eval(key, condition, super=False):
+                    expr = None
+                    annotate_query = True
+                    nonlocal person_query
+
+                    if len(condition.get("properties", {})) > 0:
+                        # Feature Flags don't support OR filtering yet
+                        target_properties = self.property_value_overrides
+                        if feature_flag.aggregation_group_type_index is not None:
+                            target_properties = self.group_property_value_overrides.get(
+                                self.cache.group_type_index_to_name[feature_flag.aggregation_group_type_index], {}
+                            )
+                        expr = properties_to_Q(
+                            Filter(data=condition).property_groups.flat,
+                            override_property_values=target_properties if not super else {},
```

Comment:
Don't really like this. Alternative is to not send override values in posthog-js decide calls needed for feature enrollment which isn't clear either

Reply:
Removed because overrides can work as long as we don't allow manual state update from posthog app which we adjusted below